Make up JSDoc for codes added/changed during the lite version of contact information development

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

Make up JSDoc for codes added/changed during the lite version of contact information development.

### Detailed test instructions:

Only JSDoc were added. Please checkout diff to see if any docs are unclear, unreadable, or confused.

### Changelog entry

>

eason9487 committed:
Make up JSDoc for codes added/changed in contact information development

ecgan commented on js/src/components/contact-information/index.js:
```diff
@@ -72,6 +76,17 @@ export function ContactInformationPreview() {
+ *
+ * @param {Object} props React props.
+ * @param {'setup-mc'|'settings'} props.view Indicate where this component is used.
+ * @param {onPhoneNumberChange} [props.onPhoneNumberChange] Called when phone number are changed.
```
```suggestion
 * @param {onPhoneNumberChange} [props.onPhoneNumberChange] Called when phone number is changed.
```

ecgan commented on js/src/components/contact-information/no-contact-information-card.js:
```diff
@@ -12,6 +12,14 @@ import Section from '.~/wcdl/section';
+ *
+ * @param {Object} props React props.
+ * @param {Function} props.onEditClick Called when clicking on the "Add information" button.
+ * @param {string} props.learnMoreUrl The URL of learn more link and the props `href` value of the link's click tracking event.
+ * @param {string} props.learnMoreLinkId The props `link_id` value of the learn more link's click tracking event.
```
```suggestion
 * @param {string} props.learnMoreUrl Value for the "Learn more" link URL and the click tracking `href` event property.
 * @param {string} props.learnMoreLinkId Value for the click tracking `link_id` event property.
```

ecgan commented on js/src/components/vertical-gap-layout/index.js:
```diff
@@ -13,6 +13,15 @@ const sizeClassName = {
+ *
+ * @param {Object} props React props.
+ * @param {string} [props.className] Additional CSS class name to be appended.
+ * @param {''|'large'|'overlap'} [props.size=''] Indicate the gap between children. The empty string as default value is a normal gap.
+ *     'large': A larger gap.
```
💅 🤔  Think it might be good to specify the value for normal gap and large gap.

eason9487 replied:
Adjusted by https://github.com/woocommerce/google-listings-and-ads/pull/968/commits/9baa3ebae5eda8aebcb0e528aa4d408472dac26e.

ecgan reviewed (approved):
LGTM 👍  Added a few suggestions in comments.

eason9487 committed:
Rephrase and correct English

Address
- https://github.com/woocommerce/google-listings-and-ads/pull/968#discussion_r695091264
- https://github.com/woocommerce/google-listings-and-ads/pull/968#discussion_r695096823

Co-authored-by: Gan Eng Chin <ecgan@users.noreply.github.com>

eason9487 committed:
Specify value for the normal gap.

Address https://github.com/woocommerce/google-listings-and-ads/pull/968#discussion_r695100450
